presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java

```diff
@@ -619,8 +628,19 @@ private synchronized void sendUpdate()
                 outputBuffers.get(),
                 totalPartitions);
         byte[] taskUpdateRequestJson = taskUpdateRequestCodec.toBytes(updateRequest);
+
+        if (taskUpdateRequestJson.length >= taskUpdateSizeLimit) {
+            throw new PrestoException(EXCEEDED_TASK_UPDATE_SIZE_LIMIT, format("TaskUpdate size exceeded the limit of %d", taskUpdateSizeLimit));
```

Comment:
nit: Maybe also include the `taskUpdateRequestJson.length` in the exception message?